### ISSUE TITLE
8.2 UEFI install issues

### ISSUE THREAD

**Issue by redowk** [xcp-ng/xcp#530]
Facing issues installing 8.2 on an Lenovo P620 (AMD Mobo/Ryzen Pro CPU) with Nvidia 1660 GPU

Seems to present be before the welcome to xcp splash screen, issue is it happens with other machines too.

Tests done:
- redownloaded, rebuilt the usb install drive;
- as a test, tried in 2 other laptops I had around (dell 7490 and lenovo e14) to see if I could boot to the install screen;
- e14 and P620 do not support legacy boot, but I was able to reach install screen on 7490 (series 8 intel cpu) using legacy boot had success;
- then tried it on the 7490 with uefi and also managed to get to the install screen.
- tried tweaking grub launch options on the p620:
set gfxpayload=keep
nomodeset
mem_encrypt=off (to disable amd memory encryption on p620)
nouveau.modeset=0

+ a bunch of other stuff, thinking it could be somehow related with the way uefi handles specific graphics on these systems.

e14 has an iris intel built in graphics
p620 has a 1660 nvidia gpu (fairly old)

obviously I don't care about the e14 but issue seems the same and could potentially be caused by the newer efi bios on this systems.

issue happens around this time (screenshot from e14):
![image](https://user-images.githubusercontent.com/6797475/145833449-5d41e905-82b5-481e-9372-d90cc8156498.png)

**Comment by redowk** [comment]
p620 screenshot before black screen with loglevel=7
![image](https://user-images.githubusercontent.com/6797475/145834422-d474bceb-7dde-448d-a9a7-64a719732c26.png)

anyway to pipe this all to a file so its not lost on reboot!?

**Comment by sajattack** [comment]
Ditto, luckily I bought a CSM mobo and booting in legacy works much better.

**Comment by enidice** [comment]
While attempting to setup PXE boot/install, I have a suspiciously similar experience on a OVH HGR-HCI-3 server containing Intel Xeon Gold 6248R - 24c/48t - 3 GHz/4 GHz processors.

I'm going to investigate BIOS mode and will report back.

![CaptureScreen (4)](https://user-images.githubusercontent.com/93122210/174260154-27da6739-f353-4c8a-9f97-dca00a8de2c3.jpeg)

**Comment by enidice** [comment]
For my case I was able to track this down to 
https://www.inspursystems.com/Product-Resources/NF5280M5/Documents/Inspur%20Server%20NF5280M5%20User%20Manual%20V2.0.pdf from:
https://www.inspursystems.com/product-resources/nf5280m5/documents/
page 110 "Advanced Power Management Configuration" --> "Power/Performance Profile" had been set to "Maximum Performance". Setting this back to the default "Custom" setting, prevented the kernel panic at this stage in the boot process. In my case it would panic in both BIOS & UEFI modes.

**Comment by olivierlambert** [comment]
Do you mean that with "Custom"  power mgmt settings you can boot and it works? Or it's just crashing later?

**Comment by enidice** [comment]
With Custom (which is actually the vendor default, but unfortunately had been changed on my first test machine) yes I'm able to boot as far as the installer menu without issue. From there, I'm successfully able to complete the installation manually (not yet attempted using answer file), and successfully able to reboot into a running system.

**Comment by dezren39** [comment]
For some users experiencing these issues, it may be a duplicate of:
- https://github.com/xcp-ng/xcp/issues/436
- There is an unofficial fix iso:
  - Forum: https://forum.level1techs.com/t/xcp-ng-intel-nuc-11/179812
  - ISO Page: https://users.ntplx.net/~andrew/xcp/
  - ISO Direct Link: https://users.ntplx.net/~andrew/xcp/xcp-ng-8.2.1-NUCtest2.iso
  - 436 mentions it will be brought into nightly, eventually.
  - This fix should work for Intel 10+ CPU's using the embedded Iris graphics.
  - The previous solution for this issue was to set CSM mode and use legacy bios.
    - Newer Intel Chipsets do not have CSM bios, only

**Comment by stormi** [comment]
No promises here, but I'd be interested in hearing from users testing with the refreshed installation ISOs for XCP-ng 8.2.1 which are currently undergoing public testing: https://xcp-ng.org/forum/topic/8026/xcp-ng-8-2-1-refreshed-installation-iso-better-hardware-support.

**Comment by rformato** [comment]
I have experienced the same thing with the latest 8.2.1 on an OVH RISE-3

**Comment by olivierlambert** [comment]
@rformato even with the new refreshed ISO?

**Comment by rformato** [comment]
@olivierlambert I downloaded today the ISO from the site, is it the refreshed one? That was my understanding

**Comment by olivierlambert** [comment]
Yes, but it brings fixes for AMD platforms, so you problem might be entirely unrelated. Please post on the community forum: https://xcp-ng.org/forum
